app/assets/javascripts/discourse/app/components/quote-button.js

```diff
@@ -104,6 +122,28 @@ export default Component.extend({
     quoteState.selected(postId, _selectedText, opts);
     this.set("visible", quoteState.buffer.length > 0);
 
+    if (this.siteSettings.enable_fast_edit) {
+      const regexp = new RegExp(quoteState.buffer, "gi");
+
+      // if we have a linebreak, the selection is probably too complex to be handled
+      // by fast edit, so ignore it
+      // if the selection is present multiple times, we also consider it too complex
+      // and ignore it, note this specific case could probably be handled in the future
+      if (
+        quoteState.buffer.length < 1 ||
+        quoteState.buffer.match(/\n/g) ||
+        postBody.match(regexp).length > 1
+      ) {
+        this.set("_isFastEditable", false);
+        this.set("_fastEditInitalSelection", null);
+        this.set("_fastEditNewSelection", null);
```

Comment:
```suggestion
        this.setProperties({
          _isFastEditable: false,
          _fastEditInitalSelection: null,
          _fastEditNewSelection: null
        });
```

Reply:
No we don't do this anymore because we will soon move to something which doesnt use set/setProperties, so one assignment by line will be easier to migrate